internal/engine/k8srollout/podmonitor.go

```diff
@@ -83,13 +87,22 @@ func (m *PodMonitor) OnChange(ctx context.Context, st store.RStore, _ store.Chan
 }
 
 func (m *PodMonitor) print(ctx context.Context, update podStatus) {
+	reusingPod := update.podStartTime.Before(update.updateFinishTime)
 	key := podManifest{pod: update.podID, manifest: update.manifestName}
-	if !m.trackingStarted[key] {
-		logger.Get(ctx).Infof("\nTracking new pod rollout (%s):", update.podID)
-		m.trackingStarted[key] = true
+	if ts, ok := m.trackingStarted[key]; !ok || ts.Before(update.updateFinishTime) {
```

Comment:
can you add some comments on what this is checking for? took me a while to figure it out

Reply:
yeah, and `trackingStarted` is really not the right name any more

Reply:
moved all the reusingPod stuff to the start of the function and undid this change